dbusmock/templates/bluez5.py

```diff
@@ -393,3 +498,625 @@ def DisconnectDevice(_self, adapter_device_name, device_address):
+        'Paired': dbus.Boolean(False, variant_level=1),
+        'Trusted': dbus.Boolean(False, variant_level=1),
+        'RSSI': dbus.Int16(-61, variant_level=1),  # arbitrary
+        'Adapter': dbus.ObjectPath(adapter_path, variant_level=1),
+        'Address': dbus.String(device_address, variant_level=1),
+        'AddressType': dbus.String("random"),
+        'Alias': dbus.String("40-A1-82-A6-BB-3D", variant_level=1),
+    }
+    if service_uuid:
+        properties['UUIDs'].append(service_uuid)
+        properties['ServiceData'] = dbus.Dictionary({service_uuid: service_data})
+    if manf_id:
+        properties['ManufacturerData'] = dbus.Dictionary({manf_id: manf_data})
+    self.AddObject(path,
+                   DEVICE_IFACE,
+                   # Properties
+                   properties,
+                   # Methods
+                   [
+                       ('CancelPairing', '', '', ''),
+                       ('Connect', '', '', ""),
+                       ('ConnectProfile', 's', '', ''),
+                       ('Disconnect', '', '', ''),
+                       ('DisconnectProfile', 's', '', ''),
+                       ('Pair', '', '', ''),
+                   ])
+
+    manager = mockobject.objects['/']
+    manager.EmitSignal(OBJECT_MANAGER_IFACE, 'InterfacesAdded',
+                       'oa{sa{sv}}', [
+                           dbus.ObjectPath(path),
+                           {DEVICE_IFACE: properties},
+                       ])
+
+
+@dbus.service.method(BLUEZ_MOCK_IFACE,
+                     in_signature='', out_signature='')
+def DeviceDiscovery(self):
+    # Eddystone URL Beacon
+    self.AddBeacon('hci0', '11:22:33:44:55:66',
+                   service_uuid='0000feaa-0000-1000-8000-00805f9b34fb',
+                   service_data=[16, 8, 1, 98, 108, 117, 101, 116, 111,
+                                 111, 116, 104, 7])
+    # Eddystone UID Beacon
+    self.AddBeacon('hci0', '11:22:33:44:55:99',
+                   service_uuid='0000feaa-0000-1000-8000-00805f9b34fb',
+                   service_data=[0, 191, 0, 0, 0, 0, 0, 69, 97, 114, 116, 104,
+                                 0, 0, 0, 0, 0, 11])
+    # AltBeacon
+    self.AddBeacon('hci0', '11:22:33:44:55:77',
+                   manf_id=65535,
+                   manf_data=[190, 172, 72, 37, 62, 89, 114, 36, 68, 99,
+                              185, 184, 3, 63, 250, 181, 202, 254, 97, 99,
+                              101, 107, 188, 0])
+    # iBeacon
+    self.AddBeacon('hci0', '11:22:33:44:55:88',
+                   manf_id=76,
+                   manf_data=[2, 21, 106, 177, 124, 23, 244, 123, 77, 65, 128,
+                              54, 82, 106, 238, 210, 47, 115, 1, 22, 3, 104,
+                              191])
+
+
+@dbus.service.method(BLUEZ_MOCK_IFACE,
+                     in_signature='ss', out_signature='')
+def ConnectMicroBit(self, adapter_name, device_address):
+    print('In connect microbit')
```

Comment:
No `print()` please, it may confuse or at least mess up the output of tests. Use `logging.debug()`.